adagrams/game.py

```diff
@@ -1,13 +1,148 @@
 from random import randint
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 
 def draw_letters():
-    pass
+    letter_bank = []
+    available_letters = build_letters_and_availability_list(LETTER_POOL)
+    
+    while len(letter_bank) < 10:
+        index_of_letter = get_index_of_letter()
 
+        if not available_letters[index_of_letter]['is_not_available']:
+            letter_bank.append(available_letters[index_of_letter]['letter'])
+            available_letters[index_of_letter]['is_not_available'] = True
+
+    reset_available_letters(available_letters)
+
+    return letter_bank
+
+def build_letters_and_availability_list(letter_pool):
+    available_letters = []
+    
+    for letter, letter_distribution in letter_pool.items():
+        for single_letter in range(letter_distribution):
+            available_letters.append({'letter':letter, 'is_not_available':False})
+    
+    return available_letters
```

Comment:
The logic here ends up creating a list with 98 dictionaries and this approach does work! I will say though if we can come up with a less nested data structure that still works to solve the problem, I'd prefer it. 

I had to pull down your code and start the debugger and debug `build_letters_and_availability_list` and `draw_letters` to see how they work together. When devs have to do this (instead of just reading the code) to understand how the logic works, it's usually a sign that something could be simplified. 

I'd poke around #code-reviews if you feel like to see some other approaches!